DZ_7_Непараметрические тесты

In [ ]:
import numpy as np
import scipy.stats as stats

Задачи № 1-4 решать с помощью функций, задачу №5 вручную \
Выбрать тест и проверить, есть ли различия между выборками:

###1. Даны две независимые выборки. Не соблюдается условие нормальности
###x1 = [380, 420, 290]
###y1 = [140, 360, 200, 900]
###Сделайте вывод по результатам, полученным с помощью функции

In [ ]:
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])

stats.mannwhitneyu(x1, y1) #используем тест Манна-Уитли

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

При α = 0.05, pvalue 0.628 > α, остается H0 - нет оснований отвергнуть нулевую гипотезу, выборки близки к  одинаковым

###2. Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?
###1е измерение до приема препарата: [150, 160, 165, 145, 155]
###2е измерение через 10 минут: [140, 155, 150, 130, 135]
###3е измерение через 30 минут: [130, 130, 120, 130, 125]

In [ ]:
a1 = np.array([150, 160, 165, 145, 155])
a2 = np.array([140, 155, 150, 130, 135])
a3 = np.array([130, 130, 120, 130, 125])
α = 0.05
stats.friedmanchisquare(a1, a2, a3) # используем критерий Фридмана, который проверяет H0 о том, что повторные измерения одних и тех же индивидов имеют одинаковое распределение.
#Он часто используется для проверки согласованности измерений, полученных разными способами.

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

При α = 0.05, pvalue 0.0083 < α, H1 - принимается, есть статистические различия групп.


###3. Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [ ]:
a1 = np.array([150, 160, 165, 145, 155])
a2 = np.array([140, 155, 150, 130, 135])
α = 0.05
stats.wilcoxon(a1, a2)
# используем Т-критерий Вилкоксона Обязательным условием для применения методики Вилкоксона является наличие двух рядов измерений. Если выборок больше, то целесообразно отказаться от Т-критерия в пользу критерия Фридмана.
#Т-критерий Вилкоксона помогает установить наличие изменений, их направление и выраженность интенсивность. С его помощью устанавливают, в каком направлении доминируют сдвиги (например, в положительную или отрицательную сторону).

WilcoxonResult(statistic=0.0, pvalue=0.0625)

При α = 0.05, pvalue 0.0625 > α, остается H0 - нет оснований отвергнуть нулевую гипотезу, без 3 измерения препарат неэффективен

###4. Даны 3 группы учеников плавания.
###В 1 группе время на дистанцию 50 м составляют: [56, 60, 62, 55, 71, 67, 59, 58, 64, 67]
###Вторая группа : [57, 58, 69, 48, 72, 70, 68, 71, 50, 53]
###Третья группа: [57, 67, 49, 48, 47, 55, 66, 51, 54]
###Сделайте вывод по результатам, полученным с помощью функции

In [ ]:
a1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
a2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
a3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
α = 0.05
stats.kruskal(a1,a2,a3) #используем критерий Крускала-Уоллиса – критерий, используемый для сравнения 3-ех и более независимых выборок по количественному или порядковому признаку, когда нарушается предположение о нормальностии
# и нужно узанть о наличии статистически значимой разницы между медианами групп.

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

При α = 0.05, pvalue 0.0650 > α, остается H0 - нет оснований отвергнуть нулевую гипотезу, группы со временем пловцов статистически равны

###5. Заявляется, что партия изготавливается со средним арифм. 2,5 см.
###Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения.
### Объем выборки 10, уровень статистической значимости 5%
[2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]

In [36]:
X_arr = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
n = len(X_arr) # =10
µ = 2.5 #µ(x) среднее арифметическое ген/совокупности
α = 0.05 / 2
df = n - 1 # =9
#Запись гипотез: H0 = X̅ = µ(x) и H1 = X̅ > µ(x)
#к вычислению X̅ = ?; std = √D = ?
X̅  = X_arr.mean()
std = X_arr.std(ddof = 1)
print (f'X̅ = {X̅} , std = {std}')

X̅ = 2.5279999999999996 , std = 0.1572542173961923


In [37]:
#в случае, когда std неизвестно, выборка менее 30, используется t-распределение
t_raschetn = (X̅ - μ) / (std / n**(1/2))
print (f't_raschetn = {t_raschetn}')

t_raschetn = 0.5630613661802959


In [39]:
#Ниже подгрузим t_critich:
t_critich = stats.t.ppf(1-α, df)
print (f't_critich = {t_critich}')

t_critich = 2.2621571627409915


In [44]:
if t_critich > t_raschetn:
    print (f'Ответ: При α = {α}, t_critich > t_raschetn, остается H0 - нет оснований отвергнуть нулевую гипотезу, партия изготавливается со средним арифм. {µ}см.')
else:
    print (f'Ответ: При α = {α}, t_critich < t_raschetn | t_critich = t_raschetn, принимаем H1 - партия изготавливается со средним арифм. более чем {µ}см.')

Ответ: При α = 0.025, t_critich > t_raschetn, остается H0 - нет оснований отвергнуть нулевую гипотезу, партия изготавливается со средним арифм. 2.5см.
